### PR TITLE
Hot reloading: add notes about the behaviour and exceptions

### PR DESCRIPTION
**Pull request by JoshVanL**
Moves hot reloading to a separate section in `component-updates.md`.

Adds notes about the behaviour of hot reloading regarding downtime and respecting init errors.

Adds exceptions to the component types ignored by hot reloading.

### CODE REVIEW

File: daprdocs/content/en/operations/components/component-updates.md [hunk 1]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+1. Unless the [`HotReload` feature gate is enabled](#hot-reloading-preview-feature), perform a [rollout restart operation](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#updating-resources) on your deployments to pick up the latest component
+
+### Self Hosted
+
+Unless the [`HotReload` feature gate is enabled](#hot-reloading-preview-feature), the process of updating a component involves a single step of stopping and restarting the `daprd` process to pick up the latest component.
+
+## Hot Reloading (Preview Feature)
 
-## Self Hosted
+> This feature is currently in [preview]({{< ref "preview-features.md" >}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:26]:
```suggestion
{{% alert title="Note" color="primary" %}}
This feature is currently in [preview]({{< ref "preview-features.md" >}}).
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 2]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+> Hot reloading is enabled by via the [`HotReload` feature gate]({{< ref "support-preview-features.md" >}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:27]:
```suggestion
Hot reloading is enabled by via the [`HotReload` feature gate]({{< ref "support-preview-features.md" >}}).
{{% /alert %}}
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 3]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+
+Dapr can be made to "hot reload" components whereby component updates are picked up automatically without the need to restart the Dapr sidecar process or Kubernetes pod.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:29]:
```suggestion
Dapr can be made to "hot reload" components, whereby component updates are picked up automatically without needing to restart the Dapr sidecar process or Kubernetes pod.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 4]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+This means creating, updating, or deleting a component manifest will be reflected in the Dapr sidecar during runtime.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:30]:
```suggestion
This means creating, updating, or deleting a component manifest is reflected in the Dapr sidecar during runtime.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 5]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+
+{{% alert title="Updating Components" color="warning" %}}
+When a component is updated it will be first closed, then re-initialized using the new configuration.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
When a component is updated, it's first closed, then re-initialized using the new configuration.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 6]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+This will cause the component to be unavailable for a short period of time during this process.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
This causes the component to be unavailable for a short period of time during this process.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 7]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+{{% /alert %}}
+
+{{% alert title="Initialization Errors" color="warning" %}}
+If the initialization processes errors when a component is created or updated through hot reloading, the Dapr sidecar will respect the component field [`spec.ignoreErrors`]({{< ref component-schema.md>}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
If the initialization processes errors when a component is created or updated through hot reloading, the Dapr sidecar respects the component field [`spec.ignoreErrors`]({{< ref component-schema.md>}}).
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 8]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+That is, the behaviour is the same as when the sidecar loads components on boot.
+- `spec.ignoreErrors=false` (*default*): the sidecar will gracefully shutdown.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
- `spec.ignoreErrors=false` (*default*): the sidecar gracefully shuts down.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 9]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+- `spec.ignoreErrors=true`: the sidecar will continue to run with neither the old or new component configuration registered.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
- `spec.ignoreErrors=true`: the sidecar will continue to run with neither the old or new component configuration registered.
```
```suggestion
- `spec.ignoreErrors=true`: the sidecar continues to run with neither the old or new component configuration registered.
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 10]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+{{% /alert %}}
 
-Unless the [`HotReload` feature gate is enabled]({{< ref "support-preview-features.md" >}}), the process of updating a component involves a single step of stopping and restarting the `daprd` process to pick up the latest component.
+All components are supported for hot reloading except for the following types.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:44]:
```suggestion
The following component types are _not supported_ for hot reloading. Any create, update, or deletion of these component types are ignored by the sidecar with a restart required to pick up changes:
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 11]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+Any create, update, or deletion of these component types will be ignored by the sidecar with a restart required to pick up changes.
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 12]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+- [Actor State Stores]({{< ref "state_api.md#configuring-state-store-for-actors" >}})
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md:46]:
```suggestion
- [Actor state stores]({{< ref "state_api.md#configuring-state-store-for-actors" >}})
```

File: daprdocs/content/en/operations/components/component-updates.md [hunk 13]
```diff
@@ -6,27 +6,45 @@ weight: 300
...
+- Workflow Backends - TODO: @joshvanl waiting on https://github.com/dapr/docs/issues/3924
```
Comment (hhunter-ms) [inline daprdocs/content/en/operations/components/component-updates.md]:
```suggestion
- Workflow backends - TODO: @joshvanl waiting on https://github.com/dapr/docs/issues/3924
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
initial review

**Comment by hhunter-ms** [comment]
a bit of this is waiting on dapr/docs#4028

**Comment by JoshVanL** [comment]
Thanks @hhunter-ms, done!

**Review by hhunter-ms — APPROVED** [review]

**Review by msfussell — APPROVED** [review]

**Pull request merged**